src/Astoneti.Microservice.Media.IntegrationTests/Controller/NewsControllerTests.cs

```diff
@@ -0,0 +1,257 @@
+{
+    public class NewsControllerTests : IDisposable
+    {
+        private readonly CustomWebApplicationFactory _factory;
+
+        public NewsControllerTests(ITestOutputHelper output)
+        {
+            _factory = new CustomWebApplicationFactory
+            {
+                Output = output
+            };
+
+            Seed();
+        }
+        public void Dispose()
+        {
+            _factory.Dispose();
+        }
+
+        private void Seed()
+        {
+            _factory.DbContext.Set<NewsEntity>().AddRange(
+                new NewsEntity
+                {
+                    Id = 1,
+                    Title = "Breaking Test News",
+                    Body = "For Test Usage",
+                    PublicationDate = DateTime.Now,
+                    
+                },
+                new NewsEntity
+                {
+                    Id = 2,
+                    Title = "Godel Mastery News",
+                    Body = "For Test Usage",
+                    PublicationDate = DateTime.Now,
+                }
+            );
+
+            _factory.DbContext.SaveChanges();
+        }
+
+        [Fact]
+        public async Task GetListAsync_ReturnsOkWithList()
+        {
+            // Arrange
+            var expectedlist = _factory.DbContext.Set<NewsEntity>().ToList();
+
+            var client = _factory.CreateClient();
+
+            // Act
+            var result = await client.GetAsync(new Uri("/news", UriKind.Relative));
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, result.StatusCode);
+
+            var resultValue = await result.Content.ReadFromJsonAsync<IList<NewsModel>>();
+
+            Assert.Equal(2, resultValue.Count);
+            for (var i = 0; i < 2; i++)
+            {
+                Assert.Equal(expectedlist[i].Id, resultValue[i].Id);
+                Assert.Equal(expectedlist[i].Title, resultValue[i].Title);
+                Assert.Equal(expectedlist[i].Body, resultValue[i].Body);
+                Assert.Equal(expectedlist[i].PublicationDate, resultValue[i].PublicationDate);
+            }
+            Assert.Equal("application/json; charset=utf-8", result.Content.Headers.ContentType?.ToString());
+        }
+
+        [Fact]
+        public async Task GetAsync_ById_ReturnsOkWithItem()
+        {
+            // Arrange
+            const int id = 1;
+            var expectedResult = _factory.DbContext.Set<NewsEntity>().First(x => x.Id == id);
+
+            var client = _factory.CreateClient();
+
+            // Act
+            var result = await client.GetAsync(new Uri("/news/1", UriKind.Relative));
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, result.StatusCode);
+            Assert.Equal("application/json; charset=utf-8", result.Content.Headers.ContentType?.ToString());
+
+            var resultValue = await result.Content.ReadFromJsonAsync<NewsModel>();
+            Assert.Equal(expectedResult.Id, resultValue.Id);
```

Comment:
it is better to check all properties of object, you can use FluentAssertion for this